cmd/query/app/grpc_handler_test.go

```diff
@@ -134,335 +148,346 @@ func newGRPCServer(t *testing.T, q *querysvc.QueryService, logger *zap.Logger, t
 	return grpcServer, lis.Addr()
 }
 
-func newGRPCClient(t *testing.T, addr net.Addr) (api_v2.QueryServiceClient, *grpc.ClientConn) {
+func newGRPCClient(t *testing.T, addr net.Addr) *grpcClient {
 	conn, err := grpc.Dial(addr.String(), grpc.WithInsecure())
 	require.NoError(t, err)
-	return api_v2.NewQueryServiceClient(conn), conn
-}
 
-func newQueryService(options querysvc.QueryServiceOptions) (*querysvc.QueryService, *spanstoremocks.Reader, *depsmocks.Reader) {
-	readStorage := &spanstoremocks.Reader{}
-	dependencyStorage := &depsmocks.Reader{}
-	return querysvc.NewQueryService(readStorage, dependencyStorage, options), readStorage, dependencyStorage
+	return &grpcClient{
+		QueryServiceClient: api_v2.NewQueryServiceClient(conn),
+		conn:               conn,
+	}
 }
 
-func initializeTestServerGRPC(t *testing.T) (*grpc.Server, net.Addr, *spanstoremocks.Reader, *depsmocks.Reader) {
-	q, readStorage, dependencyStorage := newQueryService(querysvc.QueryServiceOptions{})
-	logger := zap.NewNop()
-	tracer := opentracing.NoopTracer{}
+type queryService struct {
+	*querysvc.QueryService
+	readStorage       *spanstoremocks.Reader
+	dependencyStorage *depsmocks.Reader
+}
 
-	grpcServer, addr := newGRPCServer(t, q, logger, tracer)
+func newQueryService(options querysvc.QueryServiceOptions) queryService {
```

Comment:
this method and the queryService struct seem like unnecessary noise, I suggest just instantiating these objs inside initializeTestServerGRPCWithOptions